addons/mrp/data/mrp_data.xml

```diff
@@ -14,9 +14,10 @@
         </record>
 
         <!-- Enable the manufacturing in warehouse0 -->
-        <record id='stock.warehouse0' model='stock.warehouse'>
-            <field name='manufacture_to_resupply' eval='True'/>
-        </record>
+        <function model="stock.warehouse" name="write">
+            <value eval="[ref('stock.warehouse0')]"/>
+            <value name="vals" eval="{'manufacture_to_resupply': True}"/>
+        </function>
```

Comment:
Obscurity++

Reply:
At least it's clear that we are forcing the write :) ...

Reply:
I was referring to the facts that:
1. The write of that model is full of side effects, to records that are **not** updated by the write (basically all warehouses).
2. Setting the value is unneeded since many years (it's the default, the record entry is inside a noupdate block, thus it only kicks in at install, but it's there just for the side effects ??? :/).

With an explicit call to write (I agree in an obfuscated way) I hope people will at least take the signal to check the blame of the file to understand what's going on. Explicit comment can be added to the file.

In summary: before - inconspicuous side effects, after - it pops to the eye that something weird is going on.

Reply:
I personally prefer a comment that explains why it is written that way.  The comment may be concise, and more detailed information should be found in a commit message by blaming the file.  I find this more efficient , as it immediately reminds you the reason for the code to be that way.  This was the actual motivation behind my remark :wink:

Reply:
Another solution is to extract this "side effect" part into a private method and call it explicitly from the XML file.

Reply:
> Another solution is to extract this "side effect" part into a private method and call it explicitly from the XML file.

It's complex. The side effects parts are called interleaved with other code. I experimented extracting some parts but TBH given that MRP is involved I feel it's too complex to change without proper functional knowledge. Basically, the write method is also a _fixer_ method that ensures warehouses (and other details) are properly configured. 

Some example side effects below (non-exhaustive). They depend on how the data is in the DB. Which makes it worse: if we install stock first, change something, then install mrp.

[Side effect 1](https://github.com/odoo/odoo/blob/a3da4e6beecbb4c1104269f7087eed01d95245f5/addons/mrp/models/stock_warehouse.py#L278-L280) is trivial, and there is already an extracted method. Note the common theme: the call to `_update_location_manufacture` is done unconditionally (even if the value is the same). 

[Side effect 2](https://github.com/odoo/odoo/blob/a3da4e6beecbb4c1104269f7087eed01d95245f5/addons/stock/models/stock_warehouse.py#L155-L156) note this is for all warehouses including inactive ones. It doesn't matter what field we write.

[Side effect 3](https://github.com/odoo/odoo/blob/a3da4e6beecbb4c1104269f7087eed01d95245f5/addons/stock/models/stock_warehouse.py#L184-L247) this is the trickiest because it triggers stuff based on dependencies. The field we write here matters.

The rest of the write method has more side effects. There may be more lurking around. 

We should take the warning that this change is prone to break models that have side effects on the write. The only one that breaks in our tests is stock.warehouse. We test with all modules installed, different combinations could lead to unexpected extra side effects.

Reply:
> I personally prefer a comment that explains why it is written that way.

I added a comment.